HW2/P5/physics.pyx

```diff
@@ -113,18 +139,26 @@ cpdef update(FLOAT[:, ::1] XY,
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads, with static
         # scheduling.
-        for i in range(count):
-            sub_update(XY, V, R, i, count, Grid, grid_spacing)
+        for i in prange(count, schedule='static', chunksize=count/4, num_threads=4):
+            sub_update(XY, V, R, i, count, Grid, grid_spacing, locks)
 
         # update positions
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
-        # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+        for i in prange(count, schedule='static', chunksize=count/4, num_threads=4):
+            if XY[i,0] >= 0 and XY[i,0] <= 1 and XY[i,1] >= 0 and XY[i,1] <= 1:
+                x_coord = <unsigned int> (XY[i,0] / grid_spacing)
+                y_coord = <unsigned int> (XY[i,1] / grid_spacing)
+                Grid[x_coord, y_coord] = -1
             for dim in range(2):
                 XY[i, dim] += V[i, dim] * t
 
+        # SUBPROBLEM 2: update the grid values.
+            if XY[i,0] >= 0 and XY[i,0] <= 1 and XY[i,1] >= 0 and XY[i,1] <= 1: 
+                x_coord = <unsigned int> (XY[i,0] / grid_spacing)
+                y_coord = <unsigned int> (XY[i,1] / grid_spacing)
+                Grid[x_coord, y_coord] = i
 
```

Comment:
Elena got the point. I would really really be appreciated if you could comment more. then it would be easier for others to follow your logic/algorithm. Other than that, good job!!  Comments! Comments! Comments!